objects/src/notes/location.rs

```diff
@@ -1,74 +1,63 @@
 use alloc::string::ToString;
+use core::num::TryFromIntError;
 
 use super::{
-    ByteReader, ByteWriter, Deserializable, DeserializationError, Digest, NoteError, Serializable,
-    NOTE_TREE_DEPTH,
+    ByteReader, ByteWriter, Deserializable, DeserializationError, NoteError, Serializable,
 };
-use crate::crypto::merkle::{MerklePath, NodeIndex};
+use crate::crypto::merkle::MerklePath;
 
-/// Contains information about the origin of a note.
+/// Contains information about the location of a note.
 #[derive(Clone, Debug, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
-pub struct NoteOrigin {
+pub struct NoteLocation {
     /// The block number the note was created in.
-    pub block_num: u32,
+    block_num: u32,
 
     /// The index of the note in the note Merkle tree of the block the note was created in.
-    pub node_index: NodeIndex, // TODO: should be a u32 because the depth is always the same
+    node_index_in_block: u32,
+}
+
+impl NoteLocation {
+    pub fn block_num(&self) -> u32 {
+        self.block_num
+    }
+
+    pub fn node_index(&self) -> u64 {
+        self.node_index_in_block as u64
+    }
```

Comment:
Let's add doc comments to these methods.

Also, I wonder if the return type of `node_index()` should be `NodeIndex`. Since the depth is statically know, we should be able to create it pretty easily - but i'm not sure if this is useful somewhere downstream.

Reply:
> Also, I wonder if the return type of `node_index()` should be `NodeIndex`. Since the depth is statically know, we should be able to create it pretty easily - but i'm not sure if this is useful somewhere downstream.

Maybe I'm wrong, but I think, the depth here is useless for us.